File: Solutions/5- Palindrome Number/Solution.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x == 0)
            return true;

        if (x < 0 || x % 10 == 0)
            return false;

        int y = 0;
        while (y < x) {
            y = (y * 10) + (x % 10);
            x /= 10;
        }

        return x == y || x == (y / 10);
    }

}

File: Solutions/5- Palindrome Number/Solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True

        if x < 0 or x % 10 == 0:
            return False

        y = 0
        while y < x:
            y = (y * 10) + (x % 10)
            x //= 10

        return x == y or x == (y // 10)